Fail to build routes in Windows

Dear contributors,

Thanks for this awesome project! :)

After the recent update, I discovered voie is failing to build a route in my Windows machine. Quick jest run result is

```text
C:\Users\10629\WebstormProjects\vite-plugin-voie\packages\vite-plugin-voie>npm test

> vite-plugin-voie@0.7.1 test
> jest

 FAIL  test/routes.spec.ts
  √ given basic routes it should return the correct structure (3 ms)
  √ given dynamic routes it should return the correct structure (1 ms)
  √ given nested routes it should return the correct structure
  √ given a catch-all route it should return the correct structure (1 ms)
  √ given a glob as pagesDir and extend route it should resolve to correct path
  × given a custom root, should return the correct component path (6 ms)

  ● given a custom root, should return the correct component path

    expect(received).toEqual(expected) // deep equality

    - Expected  - 9
    + Received  + 9

      Array [
        Object {
    -     "component": "/pages/index.vue",
    -     "name": "index",
    -     "path": "/",
    +     "component": "/\\pages\\index.vue",
    +     "name": "",
    +     "path": "",
        },
        Object {
    -     "component": "/pages/user/index.vue",
    -     "name": "user",
    -     "path": "/user",
    +     "component": "/\\pages\\user\\index.vue",
    +     "name": "",
    +     "path": "",
        },
        Object {
    -     "component": "/pages/user/one.vue",
    -     "name": "user-one",
    -     "path": "/user/one",
    +     "component": "/\\pages\\user\\one.vue",
    +     "name": "",
    +     "path": "",
        },
      ]

      233 |     root,
      234 |   });
    > 235 |   expect(actual).toEqual(expected);
          |                  ^
      236 | });
      237 | 

      at Object.<anonymous> (test/routes.spec.ts:235:18)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 5 passed, 6 total
Snapshots:   0 total
Time:        1.853 s, estimated 2 s
Ran all test suites.
npm ERR! code 1
npm ERR! path C:\Users\10629\WebstormProjects\vite-plugin-voie\packages\vite-plugin-voie
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c jest

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\10629\AppData\Local\npm-cache\_logs\2021-01-07T08_27_39_339Z-debug.log
```

I believe the problem occurs after the release 0.5 and using `path.sep` inside `routes.ts` and having actual set of Vue files as a test resource would be a remedy.

--- a/packages/vite-plugin-voie/src/generator.ts
+++ b/packages/vite-plugin-voie/src/generator.ts
@@ -9,8 +9,24 @@ import { buildRoutes, stringifyRoutes } from './routes';
  */
 export async function generateRoutesCode(options: Options) {
   const { root, pagesDir, exclude, extensions, extendRoute } = options;
-  const dir = path.join(root, pagesDir);
+  const dir = normalizePath(path.join(root, pagesDir));
   const files = await resolve({ dir, extensions, exclude });
-  const routes = buildRoutes({ files, dir, extensions, root, extendRoute });
+
+  const normalizedRoot = normalizePath(root);
+  const routes = buildRoutes({
+    files,
+    dir,
+    extensions,
+    root: normalizedRoot,
+    extendRoute,
+  });
+
   return stringifyRoutes(routes, options);
+}
+
+/**
+ * Normalizes a path to use forward slashes.
+ */
+function normalizePath(str: string): string {
+  return str.replace(/\\/g, '/');
 }

--- a/packages/vite-plugin-voie/test/routes.spec.ts
+++ b/packages/vite-plugin-voie/test/routes.spec.ts
@@ -1,4 +1,3 @@
-import { join } from 'path';
 import { Route } from './../src/options';
 import { buildRoutes } from '../src/routes';
 
@@ -202,11 +201,11 @@ test('given a glob as pagesDir and extend route it should resolve to correct pat
 });
 
 test('given a custom root, should return the correct component path', () => {
-  const root = join(__dirname, 'app');
+  const root = 'app';
   const files = [
-    join(root, 'pages/index.vue'),
-    join(root, 'pages/user/index.vue'),
-    join(root, 'pages/user/one.vue'),
+    'app/pages/index.vue',
+    'app/pages/user/index.vue',
+    'app/pages/user/one.vue',
   ];
   const expected: Route[] = [
     {
@@ -228,7 +227,7 @@ test('given a custom root, should return the correct component path', () => {
 
   const actual = buildRoutes({
     files,
-    dir: join(root, 'pages'),
+    dir: 'app/pages',
     extensions: defaultExtensions,
     root,
   });

